openedx/features/qverse_features/registration/forms.py

```diff
@@ -22,3 +26,57 @@ class Meta(object):
                 'field_type': 'select'
             }
         }
+        error_messages = {
+            'registration_number': {
+                'required': 'Please enter your Registration Number.',
+            },
+            'first_name': {
+                'required': 'Please enter your First Name.',
+            },
+            'surname': {
+                'required': 'Please enter your Surname.',
+            },
+            'current_level': {
+                'required': 'Please select a valid Current Level.'
+            },
+            'department': {
+                'required': 'Please select a valid Department'
+            },
+            'programme': {
+                'required': 'Please select the valid Programme that you have been enrolled in.'
+            },
+        }
+        help_texts = {
+            'registration_number': 'Enter your registration number provided from your school.',
+            'first_name': 'Enter your first name.',
+            'surname': 'Enter your surname.',
+            'other_name': 'Enter your nick name.',
+            'current_level': 'Select your current level in school during admission.',
+            'mobile_number': 'Enter your mobile number.',
+            'department': 'Select your department.',
+            'programme': 'Select the programme that you have been enrolled in.',
+        }
+
+    def clean(self):
+        """
+        Validates that incoming user is a prospective user or not.
+
+        If optional fields of incoming user don't match with the admission data
+        we can ignore that. But the required fields of incoming user must match
+        with the admission data.
+        """
+        cleaned_data = super(QVerseUserProfileForm, self).clean()
+        try:
+            OPTIONAL_FIELDS = ['mobile_number', 'other_name']
+            prospective_user = ProspectiveUser.objects.get(
+                    registration_number__iexact=cleaned_data['registration_number'])
+            for field_name in cleaned_data.keys():
+                if field_name in OPTIONAL_FIELDS:
+                    continue
+                if str(getattr(prospective_user, field_name)).lower() != str(cleaned_data[field_name]).lower():
```

Comment:
Do we actually need to use `str` here ? wouldn't the data already be a string?

Reply:
Nope.
![image](https://user-images.githubusercontent.com/42294172/79560280-c2e3b300-80c0-11ea-9587-1844a1741ed3.png)